### DISCUSSION TITLE
How to access activity photos?

### DISCUSSION THREAD

**Discussion by zpreator** [hozn/stravalib#564] [Q&A] [answered]
Hi, first of all, just want to say I really appreciate the work done on this project! It's been a lot of fun to use in my personal projects.

I am working with version 1.6 (latest from PyPI at the time of writing).

I would like to know how to pull all the images from an activity. I am able to pull the primary image, but the rest are placeholder image urls.

I have tried the following methods:

```python
# Primary Photo (success)
activity = client.get_activity(id)
activity.photos.primary.urls["600"]
## This is the only image I can get from the activity

# Full Photos
activity = client.get_activity(id)
photo = list(activity.full_photos)[0]
photo.urls["1800"]

# Photos
activity = client.get_activity(id)
photo = list(activity.photos)[0]
photo.urls["1800"]

# Client get activity photos
activity_photos = client.get_activity_photos(id)
photo = list(activity_photos)[0]
photo.urls["1800"]
```

Here is the placeholder image that the 3 methods yielded:

![](https://d3nn82uaxijpm6.cloudfront.net/assets/media/placeholder-photo@4x-6c5d2aaeadca1292be72943c04ea6defe7dcd610da7dc87a1ccaad30e134b2d6.png)

**Comment by lwasser** [comment]
Hey @zpreator thanks for this question. We just went through a bit push with upgrading to stravalib 2. x.
We are taking a small breather for a few weeks to regroup. Someone will get back to you in the upcoming weeks (I need a small break to refuel!)

**Reply by zpreator** [reply]
Oh yeah, no rush! Thanks for all your hard work 😊. Version 2 looks amazing

**Reply by inspiran** [reply, marked as answer]
Hey, I'm normally calling stravalib with get_activity_photos(strava_activity.id, 2000) and that works great.  Be aware that URL's to placeholders are returned by Strava (e.g. placeholder for a movie clip).   

I do have code in place like:  

if 'placeholder' in fpv or 'thumbnail' in fpv:
    continue

**Reply by lwasser** [reply]
hey @zpreator @inspiran thank you for this discussion!! i just [opened a pr](https://github.com/stravalib/stravalib/pull/581) that has a small section on activity photos following this discussion. Could you review and let me know if we want to add more info, detail, etc? I welcome inline corrections, comments and edits as well. many thanks!
